src/components/icons/BoldTextIcon.tsx

```diff
@@ -1,46 +1,36 @@
-import { css, cx } from '../../../styled-system/css'
-import { icon } from '../../../styled-system/recipes'
-import { token } from '../../../styled-system/tokens'
-import IconType from '../../@types/Icon'
-import { ICON_SCALING_FACTOR } from '../../constants'
-
-/** Bold icon. */
-const BoldTextIcon = ({ fill, size = 20, style = {}, cssRaw }: IconType) => {
-  const newSize = size * ICON_SCALING_FACTOR // Ensure sizing follows scaling factor
-  const strokeColor = style.fill || fill || token('colors.fg') // Calculate stroke color
+import IconType from '../../@types/IconType'
+import AnimatedIcon from './AnimatedIcon'
+import animationData from './animations/13-bold_4.json'
 
+/** Bold Text Icon with Conditional Lottie Animation. */
```

Comment:
From this comment: https://github.com/cybersemics/em/pull/2462#pullrequestreview-2385991894

> 1. Only animate inactive → active

The Lottie icons with an active/inactive state are still being animated when going from active → inactive:

https://github.com/user-attachments/assets/21928723-4116-4930-b06a-dc9f3ef4d77a

Reply:
@raineorshine, the shortcut is missing an isActive function here.

Reply:
@trevinhofmann, it seems @raineorshine is suggesting that for shortcuts missing an isActive function, the animation should trigger whenever the button is clicked:
"For shortcuts that do not define an isActive function, then animation should be performed whenever the button is clicked."

Reply:
That’s correct. But change that to "when executed."

Reply:
@raineorshine , should the transition from bold text to normal be animated? Would this represent a state change successfully?

Reply:
The transition from bold to normal would not be animated, as that represents a transition from active → inactive.

Reply:
I believe these icons are now functioning as expected. My understanding is that the animation should trigger when activating bold/itallic/underline/strikethrough, and not when going back to normal, but I'll let @raineorshine answer your questions as well to be sure.

https://github.com/user-attachments/assets/6e94b28a-6a38-4fe2-a1e9-16a6769592cd

Reply:
Ah, there is one small issue with this still @ohrytskov.

When activating the icon, the animation plays (as expected). When deactivating the icon after the animation completes, the animation does not play (as expected).

However, when double-clicking to quickly activate and then deactivate the icon, the animation plays twice (not expected) instead of only once when activating:

https://github.com/user-attachments/assets/e111dedb-c469-4dbb-b897-744b1bc0b29f